### DISCUSSION TITLE
UI 오브젝트가 카메라를 바라볼때..

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#97] [5. Q&A]
보통 C# 에서 UI 오브젝트가 카메라를 바라볼때 다음과 같은 코드를 사용합니다. 

<img width="879" alt="스크린샷 2021-12-17 오전 12 27 07" src="https://user-images.githubusercontent.com/30133892/146399757-738e81aa-0590-49f8-a6ea-77acbe78439b.png">

이 코드를 typescript 로 변경하면 아래와 같습니다.  (아직 익숙치 않음)

<img width="802" alt="스크린샷 2021-12-17 오전 12 27 44" src="https://user-images.githubusercontent.com/30133892/146399849-41c8eefa-1571-46ad-b1a5-a7a1c8183f4c.png">

이런 코드로 만들 수 있는데..... 그러나.... 작동하지 않습니다. c# 으로는 잘 동작합니다. 

왜 그럴까요??

아시는 분 좀 알려주세요..

**Comment by x4cf** [comment]
위의 정보로는 왜 안되는지 알기 어려워보입니다.
카메라 변수 선언이나 위의 코드가 어디서 작동되는지, 스크립트가 실행되는 오브젝트의 인스팩터 등 더 많은 정보가 있다면 도와드릴 수 있을것 같습니다.

**Comment by peterkangdev** [comment]
이 코드는 실제 Canvas 오브젝트에서 작동하는 코드 입니다. 

<img width="565" alt="스크린샷 2021-12-17 오후 2 28 42" src="https://user-images.githubusercontent.com/30133892/146493918-598472df-e936-4a18-b682-82bba6594b0d.png">

전체 코드는 다음과 같습니다. 

<img width="945" alt="스크린샷 2021-12-17 오후 2 29 18" src="https://user-images.githubusercontent.com/30133892/146493959-5773a691-7911-41f2-b02e-328b787a800b.png">

**Reply by x4cf** [reply]
카메라 오브젝트가 잘 찾고있는지가 의심됩니다.
start 마지막에 콘솔로그를 찍어서 카메라를 잘 가져왔는지 확인해보시기를 추천합니다.

**Reply by mhkim0720** [reply]
카메라 트리거 엔터 안에서 찾으세요 스타트에서 찾으면 널 뱉어요

**Comment by moondory77-naverz** [comment]
안녕하세요. 단계적으로 원인을 확인하는 것이 좋을 것 같습니다.
작동이 의도하는 대로 되지 않을 때, 로그를 알려주시면 좀더 범위를 좁힐 수 있을 듯 합니다.

가장 먼저 Start() 에서 Camera 컴포넌트를 정상적으로 받아왔는지 확인이 필요할 듯 합니다.
(해당 타이밍에 GameObject.Find 를 통해, 정상적으로 Camera 컴포넌트를 가져올 수 있는 지 확인해주세요.)

좀더 단서를 제공해 주실 수 있다면, 더 구체적인 도움 드릴 수 있도록 하겠습니다.
감사합니다.
